### PR TITLE
Add support to verify touch option

### PR DESCRIPTION
**Pull request by sairamsrinivasan**
This PR adds support to verify touch options in rspec

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -256,6 +256,11 @@ RSpec.describe Article do
   it { is_expected.to embed_many(:comments) }
 end
 
+# when the touch option is provided, then we can also verify that it is set.
```
Comment (dblock) [inline README.md:259]: match other descriptions' style

"can also verify that touch is set"

**Review by dblock — CHANGES_REQUESTED** [review]
I think we want `with_touch` (assumes true), `with_touch(true)`, `with_touch(false)`. Add tests for each combination.

Questions:

1. Does `touch` appear in `with_options`? Add tests/examples/README.
2. Is touch always set/never set, meaning do we need to treat absence of `touch: true/false` differently?

**Comment by sairamsrinivasan** [comment]
> I think we want `with_touch` (assumes true), `with_touch(true)`, `with_touch(false)`. Add tests for each combination.
> 
> Questions:
> 
> 1. Does `touch` appear in `with_options`? Add tests/examples/README.
> 2. Is touch always set/never set, meaning do we need to treat absence of `touch: true/false` differently?

Great suggestion. Let me incorporate the feedback.

Does `touch` appear in `with_options`? Add tests/examples/README. -> I'll check on this.


Is touch always set/never set, meaning do we need to treat absence of `touch: true/false` differently? -> 

I think it ultimately depends on the association. Based on mongoid's documentation for `embedded_in` associations, the default value for touch is `true` as of mongoid 9. Also, the touch option can take a value other than `true/false` in the form of a symbol. This symbol refers to a field defined in the parent association.

**Comment by sairamsrinivasan** [comment]
@dblock - While implementing the suggested changes. I stumbled on a few things. Specifying the `touch` option on a `belongs_to` or `embedded_in` association requires the parent document's class to be loaded before the child. If this doesn't happen, the relatable module cannot resolve the parent class. There's a `NameError` that gets swallowed by library code here: https://github.com/mongodb/mongoid/blob/master/lib/mongoid/association/relatable.rb#L461

I worked around this by requiring the parent class in the child that defines the `belongs_to` or `embedded_in` association since this statement doesn't take this into account: https://github.com/mongoid/mongoid-rspec/blob/master/spec/spec_helper.rb#L20

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
-### 4.2.0 (Next)
+### 4.3.0 (Next)
```
Comment (dblock) [inline CHANGELOG.md:1]: Undo this, we haven't released 4.2.0.

File: lib/mongoid/rspec/version.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module Mongoid
   module RSpec
-    VERSION = '4.1.1'.freeze
+    VERSION = '4.3.0'.freeze
```
Comment (dblock) [inline lib/mongoid/rspec/version.rb:3]: This should say 4.2.0.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks nice and clean! Fix up the version change and it's good to go.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@sairamsrinivasan Any interest in helping with this gem? (see https://github.com/mongoid/mongoid-rspec/issues/237) Maybe making the next release could be a start? If yes, email me dblock at dblock dot org with your rubygems login

**Comment by sairamsrinivasan** [comment]
@dblock Thanks for reviewing and merging this PR. I’d be more than happy to help with this gem. I’ll follow up through email when I get the chance.

**Pull request merged**